extensions/sql-migration/src/wizard/skuRecommendationPage.ts

```diff
@@ -254,49 +198,74 @@ export class SKURecommendationPage extends MigrationWizardPage {
 				descriptions
 			});
 		});
-		let miDialog = new AssessmentResultsDialog('ownerUri', this.migrationStateModel, 'Assessment Dialog', this, 'mi');
-		let vmDialog = new AssessmentResultsDialog('ownerUri', this.migrationStateModel, 'Assessment Dialog', this, 'vm');
+		let miDialog = new AssessmentResultsDialog('ownerUri', this.migrationStateModel, 'Assessment Dialog', this, MigrationTargetType.SQLMI);
+		let vmDialog = new AssessmentResultsDialog('ownerUri', this.migrationStateModel, 'Assessment Dialog', this, MigrationTargetType.SQLVM);
 
 		this._rbg.onLinkClick(async (value) => {
-
-			//check which card is being selected, and open correct dialog based on link
-			if (value.description.linkDisplayValue === 'View/Change') {
-				if (value.cardId === 'AzureSQLVM') {
+			if (value.cardId === MigrationTargetType.SQLVM) {
+				this._rbg.selectedCardId = MigrationTargetType.SQLVM;
+				if (value.description.linkDisplayValue === 'View/Change') {
 					await vmDialog.openDialog();
-				} else if (value.cardId === 'AzureSQLMI') {
-					await miDialog.openDialog();
-				}
-			} else if (value.description.linkDisplayValue === 'Learn more') {
-				if (value.cardId === 'AzureSQLVM') {
+				} else if (value.description.linkDisplayValue === 'Learn more') {
 					vscode.env.openExternal(vscode.Uri.parse('https://docs.microsoft.com/en-us/azure/azure-sql/virtual-machines/windows/sql-server-on-azure-vm-iaas-what-is-overview'));
-				} else if (value.cardId === 'AzureSQLMI') {
+				}
+			} else if (value.cardId === MigrationTargetType.SQLMI) {
+				this._rbg.selectedCardId = MigrationTargetType.SQLMI;
+				if (value.description.linkDisplayValue === 'View/Change') {
```

Comment:
These display values should be localized  - although for this logic there should be a better way to determine what action to take than to compare display strings

Reply:
Currently, the component send the entire description object. This object has no ID to identify the description selected. Will create another PR for it.